src/dag.c

```diff
@@ -532,11 +530,11 @@ int RAI_parseDAGLoadArgs(RedisModuleCtx *ctx, RedisModuleString **argv,
         return -1;
       }
       RedisModule_CloseKey(key);
-      char *dictKey = RedisModule_Alloc(strlen(arg_string) + 4);
+      char *dictKey = (char*) RedisModule_Alloc((strlen(arg_string) + 5)*sizeof(char));
       sprintf(dictKey, "%s%04d", arg_string, 1);
-      AI_dictAdd(*localContextDict, (void*)dictKey, t);
+      AI_dictAdd(*localContextDict, (void*)dictKey, (void *)t);
       AI_dictAdd(*loadedContextDict, (void*)dictKey, (void *)1);
-      RedisModule_Free(dictKey);
+      // RedisModule_Free(dictKey);
```

Comment:
I don't get this, since dictAdd dup'es the key, we should also free it. Unless it doesn't and I am reading the code in dict.c incorrectly

Reply:
You're totally right Luca. Fixed in the latest commit